docs/how-to-guides/flush-qos1-events.md

```diff
@@ -0,0 +1,54 @@
+# Flush QoS1 events
+
+QoS1 events are batched and sent as a single batch to optimize the number of network calls made by the SDK. Batching comes with a side effect of event loss if the app is closed during the wait time([maxTimeBetweenTwoBatches]()) for the batch creation.
+SDK also stores all the QoS1 events in a indexedDB database, so last set of events can even be lost if the database is not persisted in next app session.
+
+Here are some ways to avoid the event loss -
+
+## Flush events on page visibility state change
+
+Clickstream Web SDK provide the [.free()](https://github.com/gojek/clickstream-web/blob/main/docs/reference/methods.md#free) which is used for cleanup purpose but in the process it also flushes all the events present in the system instantly.
+
+We can utilize `.free()` with [Page Visibility API](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API) to detect when the page goes in the `hidden` state and call the `.free()` method to flush all the events.
```

Comment:
@ashishmishra26 Say "you", instead of "we". The reader is always a single person.

Reply:
changed